version: '2.1'
networks:
  bridgednetwork:
    driver: bridge

services:
  rest:
    #image: "minority4u/revotrace:v1"
    build: restservice/
    container_name: restservice
    ports:
     - "5000:5000"
    volumes:
     - ./restservice:/code
    links:
      - mysqldb
    networks:
      - bridgednetwork
    depends_on:
      mysqldb:
        condition: service_healthy
  redis:
    image: "redis:alpine"
    container_name: redisservice
    networks:
      - bridgednetwork

  mysqldb:
    image: mysql:5.5
    container_name: mysqldb
    ports:
      - "3306:3306"
    volumes:
      - ./mysql:/docker-entrypoint-initdb.d
      - ./mysql/data:/var/lib/mysql
    networks:
      - bridgednetwork
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: usr
      MYSQL_PASSWORD: test
      MYSQL_DATABASE: sqlalchemy
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      interval: 1s
      timeout: 5s
      retries: 20
